### ISSUE TITLE
ascii2nc-provide additional support for formatting  AERONET3 

### ISSUE THREAD

**Issue by bikegeek** [dtcenter/MET#1278]
From MET Help issue:
 https://rt.rap.ucar.edu/rt/Ticket/Display.html?id=94517

Mariusz Pagowski (NOAA) needs formatting support for 870 nm AERONET3 data.  He needs this ASAP and can provide someone to work alongside MET staff to implement this.

Howard Soh has been providing information for this help desk issue, and here is his response to the request:

-------------snip-------------
It depends on an AERONET3 text file. If it has "AOD_870nm" column, it's
processed.
The sample file we have conatins AOD at 555 nm, 551 nm but not 550 nm.

asicc2nc reads AOD at following levels from the same text file below
(note: some of them are missing_value, -999):
AOD_1640nm,AOD_1020nm,AOD_870nm,AOD_865nm,AOD_779nm,AOD_675nm,AOD_667nm,
AOD_620nm,AOD_560nm,AOD_555nm,AOD_551nm,AOD_532nm,AOD_531nm,AOD_510nm,
AOD_500nm,AOD_490nm,AOD_443nm,AOD_440nm,AOD_412nm,AOD_400nm,AOD_380nm,
AOD_340nm,AOD_681nm,AOD_709nm,

Sample text file:

Date(dd:mm:yyyy),Time(hh:mm:ss),Day_of_Year,AOD_1640nm,AOD_1020nm,AOD_870nm,AOD_865nm,AOD_779nm,AOD_675nm,AOD_667nm,AOD_620nm,AOD_560nm,AOD_555nm,AOD_551nm,AOD_532nm,AOD_531nm,AOD_510nm,AOD_500nm,AOD_490nm,AOD_443nm,AOD_440nm,AOD_412nm,AOD_400nm,AOD_380nm,AOD_340nm,Precipitable_Water(cm),AOD_681nm,AOD_709nm,AOD_Empty,AOD_Empty,AOD_Empty,AOD_Empty,AOD_Empty,440-870_Angstrom_Exponent,380-500_Angstrom_Exponent,440-675_Angstrom_Exponent,500-870_Angstrom_Exponent,340-440_Angstrom_Exponent,440-675_Angstrom_Exponent[Polar],N[AOD_1640nm],N[AOD_1020nm],N[AOD_870nm],N[AOD_865nm],N[AOD_779nm],N[AOD_675nm],N[AOD_667nm],N[AOD_620nm],N[AOD_560nm],N[AOD_555nm],N[AOD_551nm],N[AOD_532nm],N[AOD_531nm],N[AOD_510nm],N[AOD_500nm],N[AOD_490nm],N[AOD_443nm],N[AOD_440nm],N[AOD_412nm],N[AOD_400nm],N[AOD_380nm],N[AOD_340nm],N[Precipitable_Water(cm)],N[AOD_681nm],N[AOD_709nm],N[AOD_Empty],N[AOD_Empty],N[AOD_Empty],N[AOD_Empty],N[AOD_Empty],N[440-870_Angstrom_Exponent],N[380-500_Angstrom_Exponent],N[440-675_Angstrom_Exponent],N[500-870_Angstrom_Exponent],N[340-440_Angstrom_Exponent],N[440-675_Angstrom_Exponent[Polar]],Data_Quality_Level,AERONET_Instrument_Number,AERONET_Site_Name,Site_Latitude(Degrees),Site_Longitude(Degrees),Site_Elevation(m)
18:09:2015,12:00:00,261,0.022949,0.031322,0.038461,-999.,-999.,0.049653,-999.,-999.,-999.,-999.,-999.,-999.,-999.,-999.,0.079107,-999.,-999.,0.097381,-999.,-999.,0.128962,0.139122,1.193526,-999.,-999.,-999.,-999.,-999.,-999.,-999.,1.399344,1.834012,1.561802,1.270403,1.382688,-999.,18,20,19,0,0,19,0,0,0,0,0,0,0,0,18,0,0,20,0,0,17,19,20,0,0,0,0,0,0,0,20,18,20,19,19,0,lev20,704,Table_Mountain,40.125000,-105.237000,1689.000000

**Comment by hsoh-u** [comment]
I don't understand what he needs.
- The interpolation AOD at 550nm wavelength from 440 and 675 (https://github.com/NCAR/MET/issues/1121) is available.
- ascii2nc does not interpolate AOD for other wavelengths. Does he need interpolated value for AOD_870nm?

**Comment by bikegeek** [comment]
Yes, he would like support for 870nm.  Sorry, I forgot that you added support for 440 and 675 for MET v9.0.  I'll update the issue

**Comment by bikegeek** [comment]
Here is the specific request from Mariusz:
--------------snip-----------
Yes, from GitHub it looks like MET only outputs 550 nm interpolated from 440 and 675 nms.
It would be good for us to output AOD at 440, 550, and 675 nms (and 870 nm in the future),
Mariusz

**Comment by hsoh-u** [comment]
MET already does this except interpolating 870 nm.

**Comment by bbakernoaa** [comment]
@TaraJensen 

As you are aware, I've wrapped the fortran TSPACK with python and created a python module that will interpolate to given values the way Mariusz has requested.  This should aid in our efforts to get MONET and MET to talk to each other through ascii2nc which is similar to #1258

**Comment by JohnHalleyGotway** [comment]
I believe this will be solved via a python embedding script which will reside in the METplus repo. And there will be no code changes to ascii2nc, or anything else in the MET repo. If this is all true, recommend that we move this issue to METplus.
